File: backend/crates/akashic-extraction/tests/fixtures/csharp/05-inheritance.cs
// EXT-6c-2: C# inheritance golden fixture
// Tests csharp_resolve_supertypes: class->interface (Implements) and class->class (Implements)
public interface IShape
{
    double Area();
}

public class Circle : IShape
{
    public double Area() { return 3.14; }
}

public class ColoredCircle : Circle
{
    public string Color() { return "red"; }
}

File: backend/crates/akashic-extraction/tests/fixtures/java/05-inheritance.java
package com.example;

/**
 * Base interface for all shapes.
 */
public interface Shape {
    double area();
}

/**
 * Drawable interface for renderable objects.
 */
public interface Drawable {
    void draw();
}

/**
 * A circle that extends no class but implements Shape and Drawable.
 */
public class Circle implements Shape, Drawable {
    private double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    @Override
    public double area() {
        return Math.PI * radius * radius;
    }

    @Override
    public void draw() {
        System.out.println("Drawing circle with radius=" + radius);
    }
}

/**
 * A colored circle that extends Circle.
 */
public class ColoredCircle extends Circle {
    private String color;

    public ColoredCircle(double radius, String color) {
        super(radius);
        this.color = color;
    }
}

/**
 * An interface that extends another interface.
 */
public interface ColoredShape extends Shape {
    String color();
}
